definitions/mobile.d.ts

```diff
@@ -6,49 +6,92 @@ declare module Mobile {
 		openDeviceLogStream(): void;
-		runApplication(applicationId: string): IFuture<void>;
-		uninstallApplication(applicationId: string): IFuture<void>;
-		listFiles(devicePath: string): IFuture<void>;
-		getFile(deviceFilePath: string): IFuture<void>;
-		putFile(localFilePath: string, deviceFilePath: string): IFuture<void>;
+	}
+	
+	interface IAndroidDevice extends IDevice {
+		adb: Mobile.IAndroidDebugBridge;		
+		sendBroadcastToDevice(action: string, extras: IStringDictionary): IFuture<number>;
 	}
 
-	interface IAppIdentifier {
+	interface IIOSDevice extends IDevice {
+		startService(serviceName: string): number;
+		mountImage(): IFuture<void>;
+		tryToExecuteFunction<TResult>(func: () => TResult): TResult;
+	}
+	
+	interface IDeviceAppDataBase {
 		appIdentifier: string;
-		deviceProjectPath: string;
-		liveSyncFormat: string;
-		encodeLiveSyncHostUri(hostUri: string): string;
-		isLiveSyncSupported(device: any): IFuture<boolean>;
-		getLiveSyncNotSupportedError(device: any): string;
 	}
 
-	interface IAndroidDevice extends IDevice {
-		debug(packageFile: string, packageName: string, debuggerSetup?: any): IFuture<void>;
+	interface IDeviceAppData extends IDeviceAppDataBase {
+		deviceProjectRootPath: string;
+		isLiveSyncSupported(device: Mobile.IDevice): IFuture<boolean>;
+	}
+	
+	interface IDeviceAppDataFactory {
+		create(appIdentifier: string, platform: string): Mobile.IDeviceAppData;
+	}
+	
+	interface IDeviceAppDataFactoryRule {
+		vanilla: any;
+		companion?: any;
+	}
+	
+	interface IDeviceAppDataProvider {
+		createFactoryRules(): IDictionary<Mobile.IDeviceAppDataFactoryRule>;
 	}
 
-	interface IIOSDevice extends IDevice {
-		startService(serviceName: string): number;
+	interface IAndroidLiveSyncService {
+		liveSyncCommands: any;
+		livesync(appIdentifier: string, liveSyncRoot: string, commands: string[]): IFuture<void>;
+		createCommandsFileOnDevice(commandsFileDevicePath: string, commands: string[]): IFuture<void>;
 	}
 
 	interface ILogcatHelper {
-		start(deviceIdentifier: string, adbPath: string): any;
+		start(deviceIdentifier: string): any;
 	}
 
 	interface ILogcatPrinter {
 		print(line: string): void;
 	}
+	
+	interface IDeviceApplicationManager {
+		getInstalledApplications(): IFuture<string[]>;
+		installApplication(packageFilePath: string): IFuture<void>;
+		uninstallApplication(appIdentifier: string): IFuture<void>; 
+		runApplication(appIdentifier: string): IFuture<void>;
+		stopApplication(appIdentifier: string): IFuture<void>;
+		restartApplication(applicationId: string): IFuture<void>;		
+	}
+	
+	interface IDeviceFileSystem {
+		listFiles(devicePath: string): IFuture<void>;
+		getFile(deviceFilePath: string): IFuture<void>;
+		putFile(localFilePath: string, deviceFilePath: string): IFuture<void>;
+		transferCollection(appIdentifier: string, localToDevicePaths: Mobile.ILocalToDevicePathData[]): IFuture<void>;
```

Comment:
There is no need to have `Collection` suffix here. Maybe `Files` will be better.